17-lifecycle-lab/src/01-cps.jsx

```diff
@@ -1,18 +1,31 @@
-/**
- * Lab 01: Click per seconds
- *
- * Write a component that counts clicks per second.
- * Component should show a button and some text, and print
- * the click-per-seconds rate.
- */
-
-import React from 'react';
-
-export default createClass({
-  render() {
-    return (<div>
-      <button>Click Fast</button>
-      <p>CPS rate: {0}</p>
-    </div>);
-  }
-});
+/**
+ * Lab 01: Click per seconds
+ *
+ * Write a component that counts clicks per second.
+ * Component should show a button and some text, and print
+ * the click-per-seconds rate.
+ */
+import React from "react";
+
+export default class CPS extends React.Component{
+	constructor(props) {
+		super(props);
+		this.state = {clicks: -1, startTime: -1}

```

Comment:
you don't really need start time as state variable since you don't update it through your application. you can put it simply on this.